### PR TITLE
Parse input stream

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: src/test/scala/net/ruippeixotog/scalascraper/browser/BrowserSpec.scala [hunk 1]
```diff
@@ -103,8 +101,7 @@ class BrowserSpec extends Specification with BrowserHelper with TestServer {
       }
 
       "parse correctly HTML from a file" in {
```
Comment (ruippeixotog) [inline src/test/scala/net/ruippeixotog/scalascraper/browser/BrowserSpec.scala]: Can you duplicate this test and make one of them use `parseFile`, so we can test both methods here? Then we have "parse correctly HTML from a file" and "parse correctly HTML from a resource" tests.

File: src/main/scala/net/ruippeixotog/scalascraper/browser/HtmlUnitBrowser.scala [hunk 2]
```diff
@@ -71,6 +73,25 @@ class HtmlUnitBrowser(browserType: BrowserVersion = BrowserVersion.CHROME) exten
...
   }
 
+  def parseInputStream(inputStream: InputStream, charset: String): Document = {
+    val response = new WebResponse(getWebResponseData(inputStream, charset), getWebRequest(charset), 0)
+    val window = newWindow()
+    HTMLParser.parseHtml(response, window)
+    HtmlUnitDocument(window)
+  }
+
+  def getWebRequest(charset: String): WebRequest = {
```
Comment (ruippeixotog) [inline src/main/scala/net/ruippeixotog/scalascraper/browser/HtmlUnitBrowser.scala]: Can you make these two methods `private[this]` and move them to the end of the class, together with the other private methods?

Regarding this method specifically, there is already a `newRequest` here. Can you either merge them or give them more distinguishable names?

File: src/main/scala/net/ruippeixotog/scalascraper/browser/Browser.scala [hunk 3]
```diff
@@ -15,6 +15,16 @@ trait Browser {
...
 
+  def parseResource(name: String, charset: String = "UTF-8"): Document = {
+    val stream = getClass.getResourceAsStream(name)
+    val document = parseInputStream(stream, charset)
+    stream.close()
+    document
+  }
+
+  /** Does not close the InputStream after reading */
+  def parseInputStream(inputStream: InputStream, charset: String = "UTF-8"): Document
```
Comment (ruippeixotog) [inline src/main/scala/net/ruippeixotog/scalascraper/browser/Browser.scala]: Just curious, why did you chose not to close the `InputStream` after reading? Do you have a specific use case in mind? If I'm not mistaken, both `Browser` implementations will always consume the entire stream, right?

**Comment by ruippeixotog** [comment]
Other than the code style suggestions, LGTM! Thanks for the contribution :)

File: src/main/scala/net/ruippeixotog/scalascraper/browser/Browser.scala (hunk 3 above)
Comment (raboof) [inline src/main/scala/net/ruippeixotog/scalascraper/browser/Browser.scala]: I didn't have any preference either way. Jsoup doesn't seem to close the `InputStream`, and generally "whoever creates a stream is responsible for closing it" makes sense in Java.

Come think of it, though, in Scala this makes less sense, and closing the `InputStream` inside scala-scraper might move some complexity from the consumer to the library - which is probably a good thing.

**Pull request merged**